controllers/configurationpolicy_controller.go

```diff
@@ -158,20 +160,47 @@ func (r *ConfigurationPolicyReconciler) PeriodicallyExecConfigPolicies(freq uint
 			skipLoop = true
 		}
 
+		scheduledPolicyCount := 0
+
+		// This is done every loop cycle since the channel needs to be variable in size to account for the number of
+		// policies changing.
+		policyQueue := make(chan *policyv1.ConfigurationPolicy, len(policiesList.Items))
+		doneQueue := make(chan struct{}, len(policiesList.Items))
```

Comment:
Did you consider using a [WaitGroup](https://pkg.go.dev/sync#WaitGroup) for this? Counting the number of policies "done" seems wasteful.

Reply:
I had forgotten about this. Thanks for the suggestion! That's much better.